IdentityCore/src/util/mac/MSIDKeychainUtil.m

```diff
@@ -75,6 +75,11 @@ - (NSString *)getTeamId
             MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Failed to retrieve team identifier. Using bundle Identifier instead.");
             NSString *bundleIdentifier = [signingDic objectForKey:(__bridge NSString*)kSecCodeInfoIdentifier];
             CFRelease(selfCode);
+            if (!bundleIdentifier)
+            {
+                // Attempt an alternate way of retreiving the bundle ID.
+                return [[NSBundle mainBundle] bundleIdentifier];
```

Comment:
>return [[NSBundle mainBundle] bundleIdentifier]; [](start = 16, length = 48)

I'm not insisting, but if you were to do as below, it would work without a bundle, too. A bundle does not have to exist.

if (!bundleIdentifier)
{
        bundleIdentifier = [[NSBundle mainBundle] bundleIdentifier];
}

if (!bundleIdentifier)
{
        // Attempt an alternate way of retreiving the bundle ID.
        bundleIdentifier = @"no-bundle-id";
}

return bundleIdentifier;

Reply:
You can also use [NSProcessInfo processInfo].processName as a fallback identifier, or require bundleId to be present. We need something dynamic to avoid collisions in the login keychain.

Reply:
If you require bundle ID to be present, it may break unsigned console test tools. In fact, this happened to us before we signed our debug builds, and some of this code previously relied on bundle being present. This wasn't a great experience, and it wouldn't be for someone playing with the SDK to assess its adoption cost. That's why I keep commenting on this. [NSProcessInfo processInfo].processName is a fine approach.

---
In reply to: [370386950](https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/674#discussion_r370386950) [](ancestors = 370386950)